tasks/ast_tools/src/derives/clone_in.rs

```diff
@@ -60,52 +62,92 @@ fn derive_enum(def: &EnumDef) -> TokenStream {
         }
     };
 
-    impl_clone_in(&ty_ident, def.has_lifetime, &alloc_ident, &body)
+    impl_clone_in(&ty_ident, def.has_lifetime, &alloc_ident, &body, &quote!())
 }
 
 fn derive_struct(def: &StructDef) -> TokenStream {
     let ty_ident = def.ident();
-
-    let (alloc_ident, body) = if def.fields.is_empty() {
-        (format_ident!("_"), quote!(#ty_ident))
+    let needs_override_clone_in_with_semantic_ids = Cell::new(false);
```

Comment:
Why do you need `Cell` here? Just a mutable var should do.

Reply:
ah, you are right, I use variable directly before, it violates borrow rules, it worked after I use `collect::<Vec<_>>`